src/Chapter2.hs

```diff
@@ -336,8 +348,11 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
-
+subList x y z = if (x < 0 || y < 1 || null z) then [] else if cutTailN y z > 0 then drop x ((iterate init z) !! cutTailN y z) else if x >= 0 then drop x z else []
```

Comment:
Wow, this is some advanced logic 😮 

I think it could be simplified. You can use `drop` and `take` functions to achieve the same. Also to check the border cases checks you can use guards 🙂 
Let me know if I need to elaborate on that.

Reply:
I was pretty sure there is something much simpler, thanks you! 🙂 I believe I managed to implement all comments and I really appreciate it.